zebrad/src/components/inbound/downloads.rs

```diff
@@ -116,12 +140,27 @@ where
 
     /// Queue a block for download and verification.
     ///
-    /// Returns true if the block was newly queued, and false if it was already queued.
+    /// Returns the action taken in response to the queue request.
     #[instrument(skip(self, hash), fields(hash = %hash))]
-    pub fn download_and_verify(&mut self, hash: block::Hash) -> bool {
+    pub fn download_and_verify(&mut self, hash: block::Hash) -> DownloadAction {
```

Comment:
Where does this return value actually get used? I'm surprised to see that changing the return type here didn't end up causing compiler errors in any other part of our code.

Reply:
The return value is not being used https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/inbound.rs#L243

The other side of the connection is not expecting anything in response to the `AdvertiseBlock` they sent to us.

I was a bit surprised about this as well but i assumed it was fine.

Reply:
@yaahc currently the `Inbound` service doesn't act on the `DownloadAction`.

But in future, we might want to return an error if a peer sends us lots of queued or cached blocks, because it means that they are a long way behind us. (Or for `zcashd`, it means their idea of our height is wrong.)